[Recognizers-Text] Port BaseTimeZoneParser methods declaration

Maihav opened the pull request:
### Changes made

This PR will:

- Add **BaseTimeZoneParser** class and the following methods:

  - parser_name
  - compute_minutes
  - convert_offset_in_mins_to_offset_string
  - convert_mins_to_regular_format
  - normalize_text
  - parse
  - filter_results
  - parse
  - get_datetime_resolution_result

<br />

- Create the **TimeZoneExtractorConfiguration** abstract class.

<br />

- Create **TimeZoneExtractorConfiguration** classes on every language (English, Spanish and French):
   
   - **EnglishTimeZoneExtractorConfiguration**
   - **SpanishTimeZoneExtractorConfiguration**
   - **FrenchTimeZoneExtractorConfiguration** 

<br />

- Update the **DateTimePeriodExtractorConfiguration** classes on each language (English, Spanish and French), adding the _timezone_extractor_ property and initializing it on the _init()_ method of them.
  
  - **EnglishDateTimePeriodExtractorConfiguration**
  - **SpanishDateTimePeriodExtractorConfiguration**
  - **FrenchDateTimePeriodExtractorConfiguration**

<br />

- Add and initialize  _time_zone_parser_  prop on Base classes:

  - **BaseDateParserConfiguration**
  - **BaseTimeZoneParser**
  - **DateTimePeriodParserConfiguration**
  - **TimeParserConfiguration**
  - **TimePeriodParserConfiguration**

<br />

- Add and initialize _time_zone_parser_  prop on **English** **ParserConfiguration** and **ExtractorConfiguration** classes:

  - **EnglishDateTimePeriodParserConfiguration**
  - **EnglishTimeParserConfiguration**
  - **EnglishTimePeriodParserConfiguration**
  - **EnglishMergedExtractorConfiguration**

<br />

- Add and initialize _time_zone_parser_  prop on **Spanish** **ParserConfiguration** class:

  - **SpanishMergedParserConfiguration**
  - **SpanishDateTimePeriodParserConfiguration**
  - **SpanishTimeParserConfiguration**
  - **SpanishTimePeriodParserConfiguration**

<br />

- Add and initialize _time_zoneextractor_  prop on **French** **ExtractorConfiguration** and **ParserConfiguration** classes:

  - **FrenchDateTimePeriodExtractorConfiguration**
  - **FrenchDateTimePeriodParserConfiguration**
  - **FrenchTimeParserConfiguration**
  - **FrenchTimePeriodParserConfiguration**

<br />

- Add and initialize _time_zone_parser_  prop on **Common** classes:

  - **EnglishCommonDateTimeParserConfiguration** 
  - **SpanishCommonDateTimeParserConfiguration** 

<br />

- Create the **TimeZoneDefinitions** class

<br />

- Create the **TimeZoneResolutionResult** class and add the following properties:
  - _value_
  - _utc_offset_mins_
  - _time_zone_text_

<br />

- Add some particular values for timezone recognition on _constants.py_ file

<br />

- Implement remove_diacritics method on QueryProcessor class

GiulianoDolceTalianoSantoro committed:
Port changes on BaseDateExtractor to Python

GiulianoDolceTalianoSantoro committed:
Merge pull request #329 from southworks/parity/date-extractor

[Python] Port the changes on BaseDateExtractor

Aliandi committed:
Revert "[Python] Port the changes on BaseDateExtractor"

Aliandi committed:
Merge pull request #331 from southworks/revert-329-parity/date-extractor

Revert "[Python] Port the changes on BaseDateExtractor"

Maihav committed:
WIP: Add new BaseTimePeriodParser class with methods

Maihav committed:
WIP add methods declaration

Aliandi commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,36 @@
+    def filter_results(self, query: str, candidate_results: List[DateTimeParseResult]) -> List[DateTimeParseResult]:
+        pass
+
+    def parse(self, extract_result: ExtractResult, ref_date: datetime) -> DateTimeParseResult:
```
You have two parse methods, this will probably break. Check if another name should be used, or maybe multiple dispatch?

Maihav replied:
In this case, I think multiple dispatch is the best option. I will be working on it.

Aliandi commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,36 @@
+    def parse(self, extract_result: ExtractResult) -> ParseResult:
+        pass
+
+    def filter_results(self, query: str, candidate_results: List[DateTimeParseResult]) -> List[DateTimeParseResult]:
```
i'm 99.9 percent sure i will ask you to kill this method in the future, but i wanna see the implementation first

Aliandi commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,36 @@
+from typing import List, Pattern
```
Check for unused imports, i dont think u are using `Pattern`

Maihav replied:
My bad

Maihav committed:
WIP: remove unused import & add default value on method

Commit:
Update base_time_zone.py

Changes
- Add stub class BaseTimeZoneExtractor

Commit:
Update base_time_zone.py

Added line

Maihav committed:
WIP: Add missing constants

Maihav committed:
WIP: Add compute_minutes implementation

Maihav committed:
WIP: Update compute_minutes

Maihav committed:
WIP: Add convert_offset_in_mins_to_offset_string implementation

Maihav committed:
WIP: Add convert_offset_in_mins_to_offset_string implementation

Maihav committed:
WIP: Add convert_mins_to_regular_format implementation & missing decorators

Maihav committed:
WIP: Add convert_mins_to_regular_format implementation & missing decorator

Maihav committed:
[Python][DateTime] Add new Specs cases on DateTime (#1976)

* Enabling python as a supported language in all languages

* Adding properties and regexes not initialized in date_extractor_config.py

* Adding properties and regexes not initialized in dateperiod_extractor_config.py

* Adding properties and regexes not initialized in datetime_extractor_config.py

* Adding properties and regexes not initialized in datetimeperiod_extractor_config.py

* Adding properties and regexes not initialized in timeperiod_extractor_config.py

* Adding properties and regexes not initialized in time_extractor_config.py

* Adding properties and regexes not initialized in set_extractor_config.py

* Adding properties and regexes not initialized in merged_extractor_config.py

* Adding properties and regexes not initialized in holiday_extractor_config.py

* Adding properties and regexes not initialized in duration_extractor_config.py

* Adding properties and regexes not initialized in merged_extractor_config.py  (Chinese)

* Adding properties and regexes not initialized in duration_extractor_config.py  (Chinese)

* Adding properties and regexes not initialized in datetimeperiod_extractor_config.py  (Chinese)

* Adding properties and regexes not initialized in datetime_extractor_config.py  (Chinese)

* Adding properties and regexes not initialized in dateperiod_extractor_config.py  (Chinese)

* Adding properties and regexes not initialized in date_extractor_config.py  (Chinese)

* Pep8 fixing

* Add relative_regex to English

* Add again all tests as not supported by python except for Chinese extractors

* Apply feedback on dateperiod_extractor.py

* Add class DateContext and method set_timex_with_context

Changes:
- Add class DateContext
- Add method set_timex_with_context to class TimexUtil

* Fix issue on DatePeriodExtractor

* Implement methods in ChineseDatePeriodExtractor

* Implement methods in ChineseMergedExtractor

* Put model failing tests as not supported by Python

* Refactor extract variable names

* Replace tuples with Dictionaries

Add some minor changes in the keys used in the dictionaries, reusing constants present in class TimeTypeConstants.

* Update specs

* Fix issue on DateTimePeriod

* Enable French tests

* Minor change in base_dateperiod.py

* Fix issue on Duration component

* Add necessary initializations after mergin

* Update base_dateperiod.py match_duration and match_within_next_prefix_regex methods

* pep8 changes

* Update utilities.py

* Add missing regexp's initializations

* Update merge_two_time_points in base_datetimeperiod.py (WIP)

* pep 8 fixing

* Update Specs

* Update BaseDatePeriodParser

- Add: get_year_context
- Edit: _merge_two_times_points
- Edit: parse

* Update ParseDatePeriodParser

Changes (WIP)
- Add: _parse_base_date_period
- Other minor changes

* Update Specs

* Update Spanish Specs on root

* Resolve Specs conflicts

* WIP Update Specs

* WIP Migrate missing cases on match_duration

* French is_connector fix

* Update match_duration()

* pep 8 fixing

* WIP change name method

* WIP improve duration method

* Minor changes on match_duration()

* Update DatePeriodParserConfiguration, EnglishDatePeriodExtractorConfiguration, EnglishDatePeriodParserConfiguration

Add decade_with_century_regex.

* Update Specs

* Minor changes on DateTimePeriod and TimePeriod components

* Update base_dateperiod.py to add method parse_decade to class BaseDatePeriodParser

* WIP Fix subarray position and regex method

* Initializing regexes and making minor functionality changes

* Update match_time_of_day  (WIP)

* pep 8 fixes

* Updated BaseDatePeriodParser, DateUtils

Add function DateUtils.int_try_parse
Advances in the implementation of methods __parse_decade and __parse_date_point_with_ago_and_later.

* Fix match_time_of_day() method (WIP)

* pep 8 fixes

* Sanitize test case input following same approach in MS#953

* Standarize Parsers test case input following same approach as Extractors

* Apply PR feedback

* Initialize regexes and update relative_duration_date() method

* Update strip_inequality_prefix() method (WIP)

* Update base_merged.py

* Enable DateTime Specs

* Update DateTime Parser Spec

* Mark as NotSupported English Model Specs

* Add non-initialize regexes

* Minor changes in base_dateperiod.py

* Update base_date.py

* pep 8 fixes

* Add Spanish DateTimeModel cases as NotSupported

* Mark DateTime French Model spec as NotSupported

* Update utilities.py

* Update add_mod() and _add_single_date_time_to_resolution() methods

* Minor changes on base_datetime and base_date

* Mark DateTime Chinese Model spec as NotSupported by Python

* Enable Test Cases for Python

Enabled Test Cases for the following inputs:

- "My vacation is from 10-1-2018-10-7-2018"
- "My vacation is from 10/1/2018 - 10/7/2018"
- "My vacation is from 10/1/2018-10/7/2018"
- "I will have a long vacation between 10/1-11/7"
- "APEC will happen in Korea Jan-Feb 2017"
- "APEC will happen in Korea Nov-Feb 2017"
- "APEC will happen in Korea Nov-Feb 5th, 2017"
- "APEC will happen in Korea Nov 18-Dec 19, 2015"
- "APEC will happen in Korea Nov 18 2014-Dec 19 2015"
- "APEC will happen in Korea on November 18-19"
- "I will leave from 5/1-5/7, 2020"
- "I will leave from 5/1-5/7/2020"
- "I will leave from 5/1/2019-5/7/2020"

* Update match_begin() and match_end()

* Initialize regexps in SpanishMergedParserConfiguration class

* Initialize regexps in FrenchMergedParserConfiguration class

* Initialize regexps in EnglishMergedParserConfiguration class

* Initialize regexps in ChineseMergedParserConfiguration class

* Update parse() on base_merged.py file

* Update parse() on base_duration.py file

* Minor functionality changes

* Update Specs

* pep 8 fixes

* Mark DateTime Chinese Model spec as NotSupported by Python

* Add decade_with_century_regex

Added to DatePeriod parsers and extractors of the languages English, French, Spanish and Chinese (WIP).

* Update DatePeriodParserConfiguration,

Changes
- Add ago_regex and later_regex to DatePeriodParserConfiguration (and affected classes of English, French, Spanish and Chinese.
- Method _parse_decade: completed the translation.
- Method __parse_date_point_with_ago_and_later (WIP).
- Small refactors

* Update to BaseDatePeriodParser

Changes
- Updated ParseDatePointWithAgoAndLater (WIP)

* Apply PR feedback

* Update base_dateperiod.py

* Minor changes in date_extractor.py

* Update base_dateperiod.py (WIP)

* Minor changes in utilities.py

* Update BasePeriodParser

Changes
- Add validation with hasattr for some of the success properties.
- Initialization of some variables.

* Update base_dateperiod.py

* Update merge_two_time_points() on base_dateperiod file and initialize necessary regexes

* Update base_dateperiod.py

Changes
Resolved failed tests cases in parsers.

* Initialize missing regexes on Spanish dateperiod_extractor_config.py file

* Minor change on base_dateperiod.py

* Update Specs

* Initialize missing regexps on Spanish dateperiod_extractor_config.py file

* Minor change on base_dateperiod.py

* Update match_duration() and merge_multiple_extractions() on base_dateperiod.py

* Minor fixes on config files

* pep8 fix

* Update Specs

* Resolve Holiday issues on BaseHolidayParser

Changes:
Using next was causing the return of the first element of the dictionary holiday_func_dictionary, instead of the relevant holiday using its holiday_key.
Resolved using get method from the Dictionary.

* Fix issues with the parsers and modify utilities

* Fix autopep

* Fix DateTimeModel Specs

* Add hyphenated support

* Update Specs

* Add missing properties on spanish parser configurtion

* Fix AssertionError with mid regex

* Update base_timeperiod.py

Changes
Changed the behavior of the end_with_valid_token flag to match .NET implementation.

* Update DateTimeModel Specs

* pep 8 fixes

* Fixes to match the master version, some changes were needed to resolve errors after resolving conflicts.

* Update DateTimeModel Specs

* Update base_dateperiod.py

* Fix SlipDateAndTime AssertionErrors

* Modify yaml_parser.py to parse bools correctly

* Minor changes in utilities file

* Update DatePeriodExtractor Specs

* Minor changes in Spanish dateperiod_extractor_config file

* Fix check_both_before_after regexp initialization

* Minor changes in base_dateperiod file

* Update Specs

* Minor changes in add_mod() and parse() method in base_merged file

* Adding updated specs

* Update BaseDatePeriodExtractor

* Updated Specs

Changes
- Fixed missing matching parenthesis, that caused failed tests in .net.
- Fixed misspelled 'jav' language identificator, also caused failed tests.

* Fix specs to match master

* Apply specs changes

* Apply PR review feedback

Grey0202 committed:
[EN DateTime] Make Q1/Q2 result generalized (.NET) (#1989)

* [EN DateTime] Make Q1/Q2 result generalized

* Marking some tests as NotSupported for Python, Java, and JS for now

aitelintII committed:
[Specs] Hindi DateTime Specs (#1990)

Maihav committed:
Finish convert_mins_to_regular_format implementation

Aliandi commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,73 @@
+            return Constants.INVALID_OFFSET_VALUE
+
+        offset_in_minutes = (hour * 60) + minutes
+        offset_in_minutes *= sign
```
why do this in two lines?

Aliandi commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,73 @@
+        if hour > Constants.HALF_DAY_HOUR_COUNT:
+            return Constants.INVALID_OFFSET_VALUE
+
+        if minutes != 0 and minutes != 15 and minutes != 30 and minutes != 45 and minutes != 60:
```
```suggestion
        if minutes not in [0,15,30,45,60]:
```

Aliandi commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,73 @@
+            if utc_offset.startswith("-"):
+                sign = Constants.NEGATIVE_SIGN  # Earlier than utc 0
+
+        hour, minutes = 0
```
```suggestion
        hour = minutes = 0
```

Aliandi commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,73 @@
+
+    @staticmethod
+    def convert_offset_in_mins_to_offset_string(self, offset_mins: int) -> str:
+        return f'UTC' + "+" if offset_mins >= 0 else "-" + self.convert_mins_to_regular_format(abs(offset_mins))
```
```suggestion
        return f'UTC {"+" if offset_mins >= 0 else "-"} {self.convert_mins_to_regular_format(abs(offset_mins))}
```

Aliandi commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,73 @@
+        tokens = list((str(offset_mins/60)).split("."))
+        hour = int(tokens[0])
+        min = int(tokens[1])
+        return str(hour) + ":" + str(min)
```
```suggestion
        return f'{hour}:{min}'
```

Commit:
Update BaseTimeZoneExtractor, Add TimeZoneExtractorConfiguration

Changes
- Applied feedback from previous PR.
- Removed unnecesary property extractor?name.
- Added required class TimeZoneExtractorConfiguration

Maihav committed:
Add normalize_text implementation

Maihav committed:
Apply PRss feedback

Maihav committed:
Add parse method implementation

Commit:
Update BaseTimeZoneExtractor: match_timezones

Changes
- First implementation of method match_timezones
- This is a work in progress.

Commit:
Update base_time_zone.py

Add line

Maihav committed:
WIP: Implement Parse method

Maihav committed:
WIP: Add TimeZoneResolutionResult and update DateTimeResolutionResult implementation

Maihav committed:
Fix autopep

Maihav committed:
Add new line at end of file

Maihav committed:
Fix autopep

Maihav committed:
Fix pep8

Aliandi committed:
[Python] Port the changes on BaseDateTimePeriodExtractor (#1986)

* Add files for DateTimePeriod parity

* Fix python test support

* Address PR review feedback

aitelint committed:
[.NET] Hindi DateTime TimePeriod support (#1991)

Maihav committed:
Implement filter_results

Maihav committed:
Add abstract method

Maihav committed:
Update resource-definitions to generate TimeZone resource

Commit:
WIP: Updates to BaseTimeZoneExtractor, QueryProcessor, and Resource Definitions

Changes
*BaseTimeZoneExtractor*
- extract
- match_location_times
- match_timezones

QueryProcessor
- remove_diacritics

Resource Definitions

Commit:
PEP8

Maihav committed:
Import TimeZoneDefinitions on init file

Maihav committed:
Add newline

Commit:
Import TimeZoneDefinitions on init file

Maihav committed:
Finish to port Parse method

Maihav committed:
Add TimeZone constant

Maihav committed:
Delete semicolon

Maihav committed:
Create TimeZoneExtractorConfiguration class

Maihav committed:
Add missing import and newline

Grey0202 committed:
[EN Datetime] Revise DatePeriod Parser (#1992)

* Revise DatePeriod Parser

* Apply review feedback

Maihav committed:
Import base_timezone

Maihav committed:
Create EnglishDateExtractorConfiguration class & Add props

Maihav committed:
WIP: Add EnglishDateExtractorConfiguration constructor & initialize props

Maihav committed:
Fix autopep

Maihav committed:
Add remove_diacritics methods on QueryProcessor class

Commit:
Updated BaseTimeZoneExtractor

Changes
- Ported completed version of method match_location_times

Commit:
Update BaseTimeZoneExtractor

Changes
- Ported method extract, that references previous ported methods.

Maihav committed:
Finish to implement EnglishDateExtractorConfiguration constructor

Commit:
Applied Feedback

Changes
- Substituted `regex.match` with `regex.finditer`

Maihav committed:
Fix naming class

Maihav committed:
Implement TimeZoneExtractorConfiguration class on Spanish

Maihav committed:
Add newline at end of file

Maihav committed:
WIP: Update DateTimePeriod on English

Maihav committed:
Finish with FrenchTimeZoneExtractorConfiguration class

Maihav committed:
Add new line

Maihav committed:
WIP: Update DateTimePeriodExtractorConfiguration on English

Maihav committed:
Unify base_time_zone & base_timezone

Maihav committed:
Merge branch 'python/timezone-migration/base-timezone-extractor-dev' into python/timezone-migration/base-timezone-extractor

Maihav committed:
Add missing extract method

Maihav committed:
Fix issue on timezone_extractor prop

Maihav committed:
Update Spanish DateTimePeriodExtractorConfig class

Maihav committed:
Update FrenchDateTimePeriodExtractorConfiguration class

aitelint committed:
[.NET] Hindi DateTime Duration support (#1993)

* Hindi DateTime Duration support

* Fixes to configuration files according to PR review

tellarin committed:
Updating entity support table

tellarin committed:
Minor: Nuget packages description updates (#1994)

Grey0202 committed:
[EN DateTimeV2]Add support for cases like "w/c" (#1996)

Maihav committed:
Add regex initialization on EnglishTimeExtractorConfiguration

Maihav committed:
Add regex initialization on EnglishTimePeriodExtractorConfiguration

Maihav committed:
Add regex initialization on SpanishTimeExtractorConfiguration

Maihav committed:
Fix import

Maihav committed:
Add regex initialization on SpanishTimePeriodExtractorConfiguration

Aliandi commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,245 @@
+            if utc_offset.startswith("-"):
+                sign = Constants.NEGATIVE_SIGN  # Earlier than utc 0
+
+        hour = minutes = 0
```
We generally use `substr in string`

Aliandi commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,245 @@
+
+    @staticmethod
+    def filter_results(self, query: str, candidate_results: List[DateTimeParseResult]) -> List[DateTimeParseResult]:
+        return candidate_results
```
Mmm does thid method do anythin?

Maihav replied:
On C# it only returns a List of DateTimeParseResult. There is no behavior implemented

Aliandi replied:
Kill it

Aliandi commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,245 @@
+
+        text = extract_result.text
+        normalized_text = self.normalize_text(text)
+        matched = (re.search(TimeZoneDefinitions.DirectUtcRegex, text)).group(2).value
```
```suggestion
        matched = TimeZoneDefinitions.DirectUtcRegex.search(text).group(2).value
```

Aliandi commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,245 @@
+
+        if offset_minutes != Constants.INVALID_OFFSET_VALUE:
+            datetime_result.value = self.get_datetime_resolution_result(offset_minutes, text)
+            datetime_result.resolution_str = Constants.UTC_OFFSET_MINS_KEY + ": " + offset_minutes
```
```suggestion
            datetime_result.resolution_str = Constants.UTC_OFFSET_MINS_KEY + ": " + offset_minutes
```
```suggestion
            datetime_result.resolution_str = f'{Constants.UTC_OFFSET_MINS_KEY}:{offset_minutes}'
```

Aliandi replied:
Use string interpolation in these cases

Aliandi commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,245 @@
+    def __init__(self, config: TimeZoneExtractorConfiguration):
+        self.config = config
+
+    def extract(self, text: str, reference: datetime = None) -> List[ExtractResult]:
```
You have two extract methods. Maybe you need multiple dispatch or one of them should not be here

Grey0202 committed:
[ES DateTimeV2] Revise date period regex (#1997)

Maihav committed:
Apply feedback: Use string interpolation

Maihav committed:
Apply feedback: Refactor search regexes

Maihav committed:
Apply feedback: Use multipledispatch on parse and extract methods

Grey0202 committed:
[EN DateTime] Minor: Add missing lunchtime YAML changes (#1999)

tellarin committed:
.NET adding internal cache prototype (perf improvement) (#1998)

* .NET adding internal cache prototype for speed (#1)

* Initial config normalization in English

* Add internal cache prototype

* Updated new Swedish recognizers and fixed Date cache experiment.

* Removing support for .NET Framework 4.5 and 4.5.2.

* Adding flag to disable internal cache.

* Fixing TurkishNumberRangeParserConfiguration to use Turkish number extractors.

* Making sure English number range passes Options internally.

* Moving cache keys to tuples. Making sure all datetime extractions receive reference time.

* Further improvements with cache reuse and key changes.

* Adding ISO 639-2 language codes to relevant pattern files

* Moving to standardize configs. Needs to be propagated to other languages.

* Reducing number of cache instances and propagating to Dutch.

* Centralizing number resultscache code and propagating configs to Chinese.

* Propagated configs and cache to Spanish and Portuguese numbers.

* Patch for other platforms after the language marker changes to match ISO

* Minor cleanup

Maihav committed:
Implement remove_diacritics method on QueryProcessor class

Maihav committed:
Add black line

Maihav committed:
Fix imports issues

Commit:
Update FrenchTimeExtractorConfiguration class

Changes:
- Add imports (fixed)
- Set up initial value for property time_zone_extractor

Aliandi committed:
[Python] Port the changes on BaseDateTimePeriodParser (#2001)

* Port changes on BaseDateExtractor to Python

* Add regexps in language configuration files and add Constants

* Variable name refactor

* Add methods in TimexUtil class, create class RangeTimexComponents and add properties in ResolutionStartEnd class

* Logic changes in BaseDateTimePeriodParser class

* merge_date_and_time_periods (logic changes)
* get_valid_connector_mod_for_date_and_time_period (new method)
* parse_date_with_period_prefix (new method)
* parse_date_with_time_suffix (new method)

* pep8 fixes

* parse_duration logic changes

* changes in parse_specific_time_of_day Spanish implementation

* Add non initialized regexps

* Update parse_specific_time_of_day and merge_two_time_points methods

* pep8 fixes

* Fix get_matched_time_range Chinese and Spanish implementation

* Logic changes in luis_time_span method in DateTimeFormatUtil

* Update merge_date_and_time_periods and parse_specific_time_of_day in ChineseDateTimePeriodParser

* Update parse_implicit_date in ChineseDateParser

* Logic changes in parse_specific_time_of_day in BaseDateTimePeriodParser

* pep8 fixes

Aliandi committed:
Merge pull request #336 from southworks/python/timezone-migration/dev

[Recognizers-Text] Update FrenchTimePeriodExtractorConfiguration class

Maihav committed:
Add missing import

Maihav committed:
Fix naming issue

Maihav committed:
Merge branch 'python/timezone-migration/base-timezone-extractor' of https://github.com/southworkscom/Recognizers-Text into python/timezone-migration/base-timezone-extractor

Maihav committed:
Add missing import on init file

Commit:
Add regex initialization on FrenchTimePeriodExtractorConfiguration

Changes:
- Add timezone_extractor property.
- Initialized property.

Maihav committed:
Add newline at end of file

Maihav committed:
Merge pull request #337 from southworks/python/timezone-migration/french-dev

[Recognizers-Text] Update FrenchTimePeriodExtractorConfiguration class

Maihav committed:
Fix properties initialization

Maihav committed:
Fix timezone_matcher initialization

Maihav committed:
Fix timezone_matcher initialization

Maihav committed:
Add parameters types and set build_matcher_from_lists to staticmethod

Maihav committed:
Polish some bits of the implementation

Maihav committed:
Add time_zone_parser prop to BaseDateParserConfiguration

Maihav committed:
Add time_zone_parser prop to EnglishCommonDateTimeParserConfiguration

Commit:
Add TimeZoneExtractor property to DateTimePeriodExtractorConfiguration

Changes:
Add TimeZoneExtractor property to DateTimePeriodExtractorConfiguration class

Commit:
Add TimeZoneExtractor property to MergedExtractorConfiguration

Changes:
Add TimeZoneExtractor property to MergedExtractorConfiguration class

Commit:
Add TimeZoneExtractor property to TimePeriodExtractorConfiguration

Changes:
Add TimeZoneExtractor property to TimePeriodExtractorConfiguration class

Commit:
Add TimeZoneExtractor property to EnglishMergedExtractorConfiguration

Changes:
Add TimeZoneExtractor property to EnglishMergedExtractorConfiguration class

Maihav committed:
Add time_zone_parser prop to DateTimePeriodParserConfiguration

Maihav committed:
Add time_zone_parser prop to TimeParserConfiguration

Maihav committed:
Add time_zone_parser prop to TimePeriodParserConfiguration

Commit:
Add time_zone_parser property to FrenchDateTimePeriodParserConfiguration

Changes:
- Add import for BaseTimeZoneParser
- Add property time_zone_parser
- Add initialization

Commit:
Add time_zone_parser property to FrenchTimeParserConfiguration

Changes:
- Add import for BaseTimeZoneParser
- Add property time_zone_parser
- Add initialization

Commit:
Add time_zone_parser property to FrenchTimePeriodParserConfiguration

Changes:
- Add import for BaseTimeZoneParser
- Add property time_zone_parser
- Add initialization

Maihav committed:
Fix return type

Maihav committed:
Add time_zone_parser prop and initialize it on EnglishDateTimePeriodParserConfiguration

Maihav committed:
Polish EnglishDateTimePeriodParserConfiguration

Commit:
Add time_zone_parser property to Spanish classes

Classes affected:
- SpanishTimePeriodParserConfiguration
- SpanishTimeParserConfiguration
- SpanishDateTimePeriodParserConfiguration

Changes:
- Add import for BaseTimeZoneParser
- Add property time_zone_parser
- Add initialization

Commit:
PEP8

Maihav committed:
Change return type on time_zone_parser

Maihav committed:
Implement abstract method on EnglishTimeParserConfiguration

Maihav committed:
Implement abstract method on EnglishTimePeriodParserConfiguration

Aliandi commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,250 @@
+                sign = Constants.NEGATIVE_SIGN  # Earlier than utc 0
+
+        hour = minutes = 0
+        if utc_offset.contains("+"):
```
```suggestion
        if '+' in utc_offset:
```

Maihav committed:
Polish compute_minutes method

Maihav committed:
Delete filter_results method

Maihav committed:
Fix import path

Commit:
Applied Feedback

Changes:
- Adjusted imports as necessary.
- Changed return type and initialization of time_zone_extractor.

Maihav committed:
Merge pull request #338 from southworks/python/timezone-migration/add-timezone-extractor-dev

[Recognizers-Text][Python] Add TimeZoneExtractor property

Maihav committed:
Implement all abstract methods

Maihav committed:
Implement time_zone_extractor abstract methods and fix naming

Maihav committed:
Implement time_zone_extractor abstract on SpanishCommonDateTimeParserConfiguration

Maihav committed:
Initialize time_zone_extractor on SpanishCommonDateTimeParserConfiguration

Maihav committed:
Fix time_zone_extractor initialization

aitelint committed:
[.NET] Hindi DateTime DatePeriod support (#2003)

aitelint committed:
[.NET] Hindi DateTime Set support (#2004)

Grey0202 committed:
[ES|PT DateTimeV2] Revise AndRegex and ConnectorAndRegex (#2002)

* Revise merge AndRegex and connectorAndRegex in Spanish

* Add support for PT and add tests

GiulianoDolceTalianoSantoro committed:
Merge branch 'master' of https://github.com/southworks/Recognizers-Text

Maihav committed:
Initialize time_zone_extractor on FrenchCommonDateTimeParserConfiguration

Maihav committed:
Delete black line

GiulianoDolceTalianoSantoro committed:
Update Specs

GiulianoDolceTalianoSantoro committed:
Merge branch 'master' of https://github.com/microsoft/Recognizers-Text

GiulianoDolceTalianoSantoro committed:
Merge branch 'master' into python/timezone-migration/base-timezone-extractor

GiulianoDolceTalianoSantoro committed:
Update utilites after merging and fix import in Spanish merged_parser_config

GiulianoDolceTalianoSantoro committed:
pep 8 fixes

GiulianoDolceTalianoSantoro committed:
Minor fix after merging

Maihav committed:
Polish EnglishTimeZoneExtractorConfiguration constructor

Maihav committed:
Delete multipledispatch implementation and add default values

Maihav committed:
Change base_time_zone.py to base_timezone.py

GiulianoDolceTalianoSantoro committed:
Enable TimeZone tests

GiulianoDolceTalianoSantoro committed:
Fix logic issues in base_timezone methods

GiulianoDolceTalianoSantoro committed:
Fix remove_diacritics method

GiulianoDolceTalianoSantoro committed:
Minor fix in build_matcher_from_list method

GiulianoDolceTalianoSantoro committed:
Update TimeZoneExtractorConfiguration files

GiulianoDolceTalianoSantoro committed:
Fix TimeZoneExtractorConfiguration constructor (WIP)

aitelint committed:
[.NET] Hindi DateTime DateTime support (#2005)

aitelint committed:
[.NET] Hindi DateTime DateTimePeriod support (#2006)

GiulianoDolceTalianoSantoro committed:
Fix TimeZoneExtractorConfiguration  init (WIP)

GiulianoDolceTalianoSantoro committed:
Refactor remove_sub_matches in utilities.py

GiulianoDolceTalianoSantoro committed:
pep 8 fixes

GiulianoDolceTalianoSantoro committed:
Fix match_location_times and refactor merge_all_tokens methods

GiulianoDolceTalianoSantoro committed:
Fix match_location_times

GiulianoDolceTalianoSantoro committed:
minor fix in remove_diacritics

aitelint committed:
[.NET] Hindi DateTime Merged partial support (#2007)

Grey0202 committed:
[EN DateTime] Support cases like "at 8.30 pm" (#2009)

GiulianoDolceTalianoSantoro committed:
fix lambda expresion in remove_sub_matches()

GiulianoDolceTalianoSantoro committed:
pep 8 fix

Maihav committed:
WIP: Update test_runner_datetime file

Maihav committed:
Merge branch 'python/timezone-migration/base-timezone-extractor' of https://github.com/southworkscom/Recognizers-Text into python/timezone-migration/base-timezone-extractor

GiulianoDolceTalianoSantoro committed:
Minor fixes in TimeZoneParser methods implementation

GiulianoDolceTalianoSantoro committed:
Minor changes in BaseTimeZoneParser parse method

GiulianoDolceTalianoSantoro committed:
Minor fix

GiulianoDolceTalianoSantoro committed:
Minor fix in the normalize_text() method

GiulianoDolceTalianoSantoro committed:
Minor change in parser() method

GiulianoDolceTalianoSantoro committed:
Fix compute_minutes method

GiulianoDolceTalianoSantoro committed:
Add tests as not supported by pyhton

GiulianoDolceTalianoSantoro committed:
Fixes for running the tests that were failling in Model component

GiulianoDolceTalianoSantoro committed:
Revert "Fixes for running the tests that were failling in Model component"

This reverts commit 7ca52c5e98e93e5765a127265f5f2efdb204e9c8.

GiulianoDolceTalianoSantoro committed:
Fixes for running the tests that were failling in Model component

Aliandi committed:
Merge branch 'python/timezone-migration/base-timezone-extractor' of https://github.com/southworks/Recognizers-Text into python/timezone-migration/base-timezone-extractor
